node/src/accountant/db_access_objects/receivable_dao.rs

```diff
@@ -279,43 +281,71 @@ impl ReceivableDaoReal {
-                let overflow_update_clause = "update receivable set balance_high_b = :balance_high_b, balance_low_b = :balance_low_b, \
+        let overflow_update_clause = "update receivable set balance_high_b = :balance_high_b, balance_low_b = :balance_low_b, \
                 last_received_timestamp = :last_received where wallet_address = :wallet";
 
-                self.big_int_db_processor.execute(
-                    Either::Right(&xactn),
-                    BigIntSqlConfig::new(
-                        main_sql,
-                        overflow_update_clause,
-                        SQLParamsBuilder::default()
-                            .key(WalletAddress(&transaction.from))
-                            .wei_change(Subtraction("balance", transaction.wei_amount))
-                            .other_params(vec![Param::new(
-                                (":last_received", &to_time_t(timestamp)),
-                                true,
-                            )])
-                            .build(),
-                    ),
-                )?
+        received_payments.iter().for_each(|received_payment| {
+            let last_received = to_time_t(timestamp);
+            let params = SQLParamsBuilder::default()
+                .key(WalletAddress(&received_payment.from))
+                .wei_change(Subtraction("balance", received_payment.wei_amount))
+                .other_params(vec![Param::BothClauses((":last_received", &last_received))])
+                .build();
+
+            let write_result = big_int_db_processor.execute(
+                conn,
+                BigIntSqlConfig::new(main_sql, overflow_update_clause, params),
+            );
+
+            if let Err(e) = write_result {
+                match e {
+                    BigIntDbError::General(err_msg) => panic!("{}", err_msg),
+                    BigIntDbError::RowChangeMismatch { .. } => {
+                        Self::log_or_panic(conn, received_payment, logger)
+                    }
+                }
             }
+        })
+    }
+
+    fn log_or_panic(
+        conn: &dyn ConnectionWrapper,
+        received_payment: &BlockchainTransaction,
+        logger: &Logger,
+    ) {
+        if Self::check_row_presence(conn, &received_payment.from) {
+            panic!(
+                "Update for received payment with {} wei was run without producing a data \
+            change, despite the account for wallet address {} is present",
+                received_payment.wei_amount, received_payment.from
+            )
+        } else {
+            info!(
+                logger,
+                "Received a transaction with {} wei from address {} that does not belong to any of \
+                the known debtors. Ignoring",
+                received_payment.wei_amount,
+                received_payment.from
+            )
         }
-        match xactn.commit() {
```

Comment:
Why don't we need to commit now?

Reply:
We will...plans have changed. Now, after a discussion with Dan, we want to make use of the `commit()` byproduct which is a rollback (for all txs) which would happen before we update the start block. That way we can recover from panics without serious accountancy errors.